src/formatting/reorder.rs

```diff
@@ -268,6 +282,38 @@ fn contains_macro_use_attr(attrs: &[ast::Attribute]) -> bool {
     crate::formatting::attr::contains_name(attrs, sym::macro_use)
 }
 
+/// Divides imports into three groups, corresponding to standard, external
+/// and local imports. Sorts each subgroup.
+fn group_and_sort_imports(uts: Vec<UseTree>) -> Vec<Vec<UseTree>> {
+    let mut std_imports = Vec::new();
+    let mut external_imports = Vec::new();
+    let mut local_imports = Vec::new();
+
+    for ut in uts.into_iter() {
+        if ut.path.is_empty() {
```

Comment:
I'm assuming this should never happen? Same for having a Glob or List as the first element (line 306). I can panic here instead if this is true.

Reply:
Unless it substantially impacts the implementation, I'd prefer to add the new feature without introducing panics so I think the check and continue here is fine :+1: 

There's only a couple explicit panics within rustfmt, but these are to address programmer errors (like a rustfmt developer calculating a bad span), and while there's a handful of `unreachable` cases, we do see users report panics from time to time. Better safe than sorry here :smile: